framework/helpers/BaseStringHelper.php

```diff
@@ -462,12 +462,12 @@ public static function mb_ucfirst($string, $encoding = 'UTF-8')
      */
     public static function mb_ucwords($string, $encoding = 'UTF-8')
     {
-        if (empty($string)) {
+        if ($string === null || $string === '') {
             return '';
         }
 
         $parts = preg_split('/(\s+[^\w]+\s+|\s+)/u', (string) $string, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
-        $upEven = !trim($parts[0]);
+        $upEven = !trim($parts[0][0]);
```

Comment:
What's "up even"?

Reply:
if first value is "whitespace", then converts only even values, else converts only odd values.

Reply:
Is it more performant than calling `mb_ucfirst` on each array element?

Reply:
@samdark as I'm already say in #18796:
> and it work faster: current: ~0,16s, new ~0,13s,